CardsCashCasino/Data/HoldEmUtil.cs

```diff
@@ -0,0 +1,335 @@
+                }
+            }
+            return pairs == 2;
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a pair.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a pair, false otherwise.</returns>
+        private static bool IsPair(List<Card> cards)
+        {
+            for (int i = 1; i < cards.Count; i++)
+            {
+                if (cards[i].SameVal(cards[i - 1]))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Call each of the functions used to determine if our hand is of a certain rank back to back. 
+        /// Requires that cards is of length 5, and has had SortedHand called on it.
+        /// </summary>
+        /// <param name="cards">The hand we are trying to determine the rank of.</param>
+        /// <returns>the cards rank.</returns>
+        private static Ranking GetRanking(List<Card> cards)
+        {
+            Debug.Assert(cards.Count == 5);
+
+            if (IsRoyalFlush(cards)) return Ranking.ROYAL_FLUSH;
+            else if (IsStraightFlush(cards)) return Ranking.STRAIGHT_FLUSH;
+            else if (IsFourOfAKind(cards)) return Ranking.FOUR_OF_A_KIND;
+            else if (IsFullHouse(cards)) return Ranking.FULL_HOUSE;
+            else if (IsFlush(cards)) return Ranking.FLUSH;
+            else if (IsStraight(cards)) return Ranking.STRAIGHT;
+            else if (IsThreeOfAKind(cards)) return Ranking.THREE_OF_A_KIND;
+            else if (IsTwoPair(cards)) return Ranking.TWO_PAIR;
+            else if (IsPair(cards)) return Ranking.PAIR;
+            else return Ranking.HIGH_CARD;
+        }
+
+        /// <summary>
+        /// Orders cards in ascending order by value and suit.
+        /// Suits are ordered as SPADES < HEARTS < DIAMONDS < CLUBS.
+        /// the hand [4C, 3S, 4H, AS, 7D]  would be ordered as [3S, 4H, 4C, 7D, AS]
+        /// </summary>
+        /// <param name="cards">Represents a hand of cards. There should be exactly 5 elements in the list.</param>
+        /// <returns>The sorted list of cards. There will be exactly 5 elements in the list.</returns>
+        private static List<Card> SortedHand(List<Card> cards)
+        {
+            return cards.OrderBy(card => (card.GetPokerValue() * 4) - card.Suit).ToList();
+        }
+
+        /// <summary>
+        /// Takes in a list of cards of length > 3 (probably 5 when called) and generates a list of every unique set of 3 cards.
+        /// i.e. for cards [2C,JH,5C,6S,KD], the 3-permutation [2C,JH,5C] will be returned, but never [JH,5C,2C].
+        /// </summary>
+        /// <param name="list">List of cards with length > 3</param>
+        /// <returns>All 3-card permutations of the input list</returns>
+        private static IEnumerable<IEnumerable<Card>> Get3CardPermutations(List<Card> list)
```

Comment:
passing this around as an interface is scary. Instead, get cast it into a list using Linq's "toList()" method.